### DISCUSSION TITLE
Versioning for apps NOT libs

### DISCUSSION THREAD

**Discussion by lr001dev** [nrwl/nx#6437] [Feature Requests]
We currently have a lerna workflow to version our libraries and currently working well but not sure what we need to do to version different apps in the workspace. Is there a workflow to achieve this?

**Comment by CaffeinatedCodeMonkey** [comment]
I tried to use Lerna to handle versioning, but found that it wasn't reliable. Lerna and NX didn't agree on what had actually changed in all cases. NX has its own dependency graph to figure out when something has changed whereas it seems Lerna relies on changes to the package.json  in order to recognize when changes in a different project impact the versionable application. I ended up switching to https://github.com/jscutlery/semver. It also required fewer special configurations (like telling it where to find the dist for a library, in order to publish it).
